coalib/bearlib/__init__.py

```diff
@@ -82,34 +100,56 @@ def _deprecate_decorator(func):
         logged_deprecated_args = set()
 
         def wrapping_function(*args, **kwargs):
-            for arg, depr_arg_and_modifier in depr_args.items():
-                deprecated_arg, _func = (
-                    depr_arg_and_modifier
-                    if isinstance(depr_arg_and_modifier, tuple)
-                    else (depr_arg_and_modifier, lambda x: x))
-                if deprecated_arg in kwargs:
-                    if deprecated_arg not in logged_deprecated_args:
-                        logging.warning(
-                            'The setting `{}` is deprecated. Please use `{}` '
-                            'instead.'.format(deprecated_arg, arg))
-                        logged_deprecated_args.add(deprecated_arg)
-                    depr_arg_value = _func.__call__(kwargs[deprecated_arg])
-                    if arg in kwargs and depr_arg_value != kwargs[arg]:
-                        logging.warning(
-                            'The value of `{}` and `{}` are conflicting.'
-                            ' `{}` will be used instead.'.format(
-                                  deprecated_arg, arg, arg))
+            for arg, depr_args_and_modifier in depr_args.items():
+                if not isinstance(depr_args_and_modifier, tuple):
+                    depr_args_and_modifier = (depr_args_and_modifier,)
+
+                if callable(depr_args_and_modifier[-1]):
+                    deprecated_args = depr_args_and_modifier[:-1]
+                    _func = depr_args_and_modifier[-1]
+                else:
+                    deprecated_args = depr_args_and_modifier
+                    _func = _do_nothing
+
+                for deprecated_arg in deprecated_args:
+                    if isinstance(deprecated_arg, dict):
+                        modifier = list(deprecated_arg.values())[0]
+                        deprecated_arg = list(deprecated_arg.keys())[0]
                     else:
-                        kwargs[arg] = depr_arg_value
-                    del kwargs[deprecated_arg]
+                        modifier = _func
+                    if deprecated_arg in kwargs:
+                        if deprecated_arg not in logged_deprecated_args:
+                            logging.warning(
+                                'The setting `{}` is deprecated. Please use'
+                                ' `{}` instead.'.format(deprecated_arg, arg))
+                            logged_deprecated_args.add(deprecated_arg)
+                        depr_arg_value = modifier.__call__(
+                            kwargs[deprecated_arg])
+                        if arg in kwargs and depr_arg_value != kwargs[arg]:
+                            logging.warning(
+                                'The value of `{}` and `{}` are conflicting.'
```

Comment:
there is no test coverage of the 'are conflicting' message.  Could you please add it as a doctest in the docstring